### ISSUE TITLE
JDK12 b14 java.lang.NoSuchFieldException: override

### ISSUE THREAD

**Issue by sdfelts** [jbossas/jboss-classfilewriter#18]
Caused By: java.lang.NoSuchFieldException: override
	at java.base/java.lang.Class.getDeclaredField(Class.java:2412)
	at org.jboss.classfilewriter.ClassFile$1.run(ClassFile.java:392)
	at org.jboss.classfilewriter.ClassFile$1.run(ClassFile.java:385)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at org.jboss.classfilewriter.ClassFile.<clinit>(ClassFile.java:385)
	Truncated. see log file for complete stacktrace

**Comment by sdfelts** [comment]
Immediate fix is to use setAccessible until things are locked down.

**Comment by sdfelts** [comment]
The same problem also occurred in weld-core.
org/jboss/weld/util/bytecode/ClassFileUtils

**Comment by rsvoboda** [comment]
I see the same issue when when running WF build + TS on JDK 12 (WildFly Test Suite: Integration - Smoke) - org.jboss.as.test.smoke.webservices.WSTestCase
```
Caused by: java.lang.NoSuchFieldException: override
	at java.base/java.lang.Class.getDeclaredField(Class.java:2410)
	at org.jboss.classfilewriter@1.2.3.Final//org.jboss.classfilewriter.
                    ClassFile$1.run(ClassFile.java:392)
```

@stuartwdouglas @dmlloyd 

https://github.com/jbossas/jboss-classfilewriter/blob/master/src/main/java/org/jboss/classfilewriter/ClassFile.java#L392

**Comment by rsvoboda** [comment]
Plus there are twelve `Could not initialize class org.jboss.classfilewriter.ClassFile` errors in smoke module.

```
Caused by: java.lang.NoClassDefFoundError: Could not initialize class org.jboss.classfilewriter.ClassFile
	at org.jboss.as.ejb3@15.0.0.CR1//org.jboss.as.ejb3.iiop.handle.SerializationHackProxy.proxy(SerializationHackProxy.java:55)
	at org.jboss.as.ejb3@15.0.0.CR1//org.jboss.as.ejb3.iiop.handle.HandleDelegateImpl.<init>(HandleDelegateImpl.java:58)
	at org.jboss.as.ejb3@15.0.0.CR1//org.jboss.as.ejb3.deployment.processors.IIOPJndiBindingProcessor.bindService(IIOPJndiBindingProcessor.java:108)
	at org.jboss.as.ejb3@15.0.0.CR1//org.jboss.as.ejb3.deployment.processors.IIOPJndiBindingProcessor.deploy(IIOPJndiBindingProcessor.java:83)
	at org.jboss.as.server@7.0.0.Beta1//org.jboss.as.server.deployment.DeploymentUnitPhaseService.start(DeploymentUnitPhaseService.java:144)
	... 8 more
```

**Comment by rsvoboda** [comment]
@manovotn / @mkouba fyi:
> The same problem also occurred in weld-core.
> org/jboss/weld/util/bytecode/ClassFileUtils

**Comment by manovotn** [comment]
Thanks for the heads-up Rostislav, haven't checked JDK 12 and Weld quite yet.

**Comment by manovotn** [comment]
> Immediate fix is to use setAccessible until things are locked down.

You cannot use `setAccessible` for you will get an exception trying to invoke a protected method(`defineClass()`) on a CL. That's why Unsafe was used in the first place. Unless I am missing something, which might well be the case :)

**Comment by sdfelts** [comment]
Until things are locked down, changing to use setAccessible works fine.
After things are locked down, it will be necessary to open packages but it will still provide something that works.
The current code doesn't work at all.  It's a simple change.
                    // make both accessible
                    defineClass1.setAccessible(true);
                    defineClass2.setAccessible(true);

**Comment by manovotn** [comment]
That's what was there until 1.2.1.Final but was deliberately changed. I don't think we want to just revert that.

**Comment by rsvoboda** [comment]
@bstansberry @kabir if WF 16 going to tackle Java 12 ? 
https://openjdk.java.net/projects/jdk/12/ - 2019/03/19 General Availability

btw there is already http://jdk.java.net/13/

**Comment by ropalka** [comment]
This was fixed with commit f08463697c5df6d1f5ca061056825cdb11cc2a34
